plugins/package-managers/nuget/src/main/kotlin/utils/NuGetInspector.kt

```diff
@@ -0,0 +1,304 @@
+        val errorMessage = collectErrorMessage(result)
+        val issues = if (errorMessage.isNotBlank()) {
+            listOf(
+                Issue(
+                    message = errorMessage,
+                    source = "NugetInspector"
+                )
+            )
+        } else {
+            emptyList()
+        }
+        return listOf(ProjectAnalyzerResult(project, packages, issues))
+    }
+
+    @Serializable
+    internal data class Result(
+        val packages: List<PackageData>,
+        val dependencies: List<PackageData>,
+        val headers: List<Header>
+    )
+
+    @Serializable
+    internal data class Header(
+        @SerialName("project_framework") val projectFramework: String,
+        val errors: List<String>
+    )
+
+    @Serializable
+    internal data class PackageData(
+        val type: String,
+        val namespace: String?,
+        val name: String,
+        val version: String?,
+        val description: String,
+        val parties: List<Party>,
+        @SerialName("homepage_url") val homepageUrl: String?,
+        @SerialName("download_url") val downloadUrl: String,
+        val size: Long,
+        val sha1: String?,
+        val md5: String?,
+        val sha256: String?,
+        val sha512: String?,
+        @SerialName("code_view_url") val codeViewUrl: String?,
+        @SerialName("vcs_url") val vcsUrl: String?,
+        val copyright: String?,
+        @SerialName("license_expression") val licenseExpression: String?,
+        @SerialName("declared_license") val declaredLicense: String?,
+        @SerialName("source_packages") val sourcePackages: List<String>,
+        @SerialName("repository_homepage_url") val repositoryHomepageUrl: String?,
+        @SerialName("repository_download_url") val repositoryDownloadUrl: String?,
+        @SerialName("api_data_url") val apiDataUrl: String,
+        val purl: String,
+        val dependencies: List<PackageData>,
+        val errors: List<String>,
+        val warnings: List<String>
+    )
+
+    @Serializable
+    internal class Party(
+        val type: String,
+        val role: String,
+        val name: String?,
+        val email: String?,
+        val url: String?
+    )
+}
+
+private const val TYPE = "NuGet"
+
+private fun List<NuGetInspector.Party>.toAuthors(): SortedSet<String> =
+    filter { it.role == "author" }.mapNotNullTo(sortedSetOf()) { party ->
+        buildString {
+            party.name?.let { append(it) }
+            party.email?.let {
+                append(if (party.name != null && it.isNotBlank()) " <$it>" else it)
+            }
+        }.takeIf { it.isNotBlank() }
+    }
+
+private fun collectErrorMessage(result: NuGetInspector.Result): String {
+    var errorMessage = ""
+    result.headers.first().errors.forEach {
+        errorMessage += it + "\n"
+    }
+    result.packages.first().errors.forEach {
+        errorMessage += it + "\n"
+    }
```

Comment:
If `errors` are individual errors and not lines of the message for the same error, each errors should be mapped to a separate ORT issue, and not everything be merged into a single string.

Reply:
In our cases these are not errors that can handled differently.

Reply:
It does not matter whether the error could be handled differently / separately. What matters is if the strings are (semantically) different errors, or just different message lines of the semantically the same error.

Reply:
ack. So we are OK here as it is.

Reply:
So you confirm these to be just lines of a single message that belongs to a single error?

Reply:
yes

Reply:
Sorry, but I still believe that's not correct. See:

    result.dependencies.forEach { dep -> dep.errors.forEach { errorMessage += it + "\n" } }

This iterates over all dependencies, and concats all their errors into a single string. But different dependencies might have errors for completely different reasons. IMO these errors should have been mapped to the `issues` of a `PackageReference` instead.